pkg/scheduler/worker/workerpool_test.go

```diff
@@ -87,6 +88,63 @@ func TestWorkerPool(t *testing.T) {
 	requireOpsProcessableExists(t, opsProcessable, testInvoker.params[0].CorrelationID)
 }
 
+func TestWorkerPoolMaxOpRetriesReached(t *testing.T) {
+	test.IntegrationTest(t) //required because a valid Kubeconfig is required to create test cluster entry
+
+	//create cluster inventory
+	inventory, err := cluster.NewInventory(db.NewTestConnection(t), true, &cluster.MetricsCollectorMock{})
+	require.NoError(t, err)
+
+	//add cluster to inventory
+	clusterState, err := inventory.CreateOrUpdate(1, kebTest.NewCluster(t, "1", 1, false, kebTest.OneComponentDummy))
+	require.NoError(t, err)
+
+	//cleanup created cluster
+	defer func() {
+		require.NoError(t, inventory.Delete(clusterState.Cluster.RuntimeID))
+	}()
+
+	//create test invoker to be able to verify invoker calls
+	testInvoker := &testInvoker{}
+
+	//create reconciliation for cluster
+	testInvoker.reconRepo = reconciliation.NewInMemoryReconciliationRepository()
+	_, err = testInvoker.reconRepo.CreateReconciliation(clusterState, nil)
+	require.NoError(t, err)
+	opsProcessable, err := testInvoker.reconRepo.GetProcessableOperations(0)
```

Comment:
Can we test with `defaultMaxParallelOperations`? which is 25, not 0

Reply:
Yup! Have updated the code 👍

Reply:
done